### PR TITLE
[ACA-2219] support more precise searching

### PR DESCRIPTION
**Pull request by DenysVuika**
## PR Checklist
Please check if your PR fulfills the following requirements:

```
- [x] The commit message follows our guidelines: https://github.com/Alfresco/alfresco-content-app/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
```

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application / Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [ACA-2219](https://issues.alfresco.com/jira/browse/ACA-2219)

## What is the new behavior?

Support for more precise search queries that include conjunctions, disjunctions, and phrases.
See more details in the corresponding [ticket](https://issues.alfresco.com/jira/browse/ACA-2219)

screenshot 2019-02-26 at 13 52 43

screenshot 2019-02-26 at 14 14 17

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/components/search/search-results/search-results.component.ts [hunk 1]
```diff
@@ -114,21 +114,62 @@ export class SearchResultsComponent extends PageComponent implements OnInit {
...
+    }
+    return false;
+  }
+
+  private formatFields(fields: string[], term: string): string {
+    let prefix = '';
+    let suffix = '*';
+
+    if (term.startsWith('=')) {
+      prefix = '=';
```
Comment (pionnegru) [inline src/app/components/search/search-results/search-results.component.ts]: what about prohibit operator `-` ,  required operator `+` ?

File: src/app/components/search/search-results/search-results.component.spec.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { NavigateToFolder } from '../../../store/actions';
 import { Pagination } from '@alfresco/js-api';
 import { SearchQueryBuilderService } from '@alfresco/adf-content-services';
 
-describe('SearchComponent', () => {
+fdescribe('SearchComponent', () => {
```
Comment (suzanadirla) [inline src/app/components/search/search-results/search-results.component.spec.ts]: ups :)

Comment (DenysVuika) [reply src/app/components/search/search-results/search-results.component.spec.ts]: thanks, fixed

File: src/app/components/search/search-results/search-results.component.ts (hunk 1 above)
Comment (DenysVuika) [reply src/app/components/search/search-results/search-results.component.ts]: I think that will go into a separate task/ticket. There's much more stuff to support in the long run https://docs.alfresco.com/6.0/concepts/rm-searchsyntax-intro.html

**Pull request merged**